TreeModelLib/AbovegroundCompetition/SimpleAsymmetricZOI/SimpleAsymmetricZOI.py

```diff
@@ -0,0 +1,156 @@
+        self.canopy_height[np.where(np.less(self.canopy_height, 0))] = 0
+
+        #define array to count wins
+        wins = np.zeros_like(my_height)
+        #indicate, where tree is highest
+        wins[np.where(np.equal(my_height,
+                               self.canopy_height[:, :, np.newaxis]))] += 1
+
+        #Account for shared wins
+        cumwins = wins.sum(axis=-1)
+        cumwins[np.where(cumwins == 0)] = -1e-4
+        wins = wins / cumwins[:, :, np.newaxis]
+        wins[np.where(wins < 0)] = 0
+        #Count number of wins
+        wins = wins.sum(axis=(0, 1))
+        self.aboveground_resources = wins / crown_areas
+
+    ## This function calculates the tree height at a (mesh-)point depending
+    #  on the distance from the tree position.\n
+    #  @param stem_height - stem height\n
+    #  @param crown_radius - crown radius\n
+    #  @param distance - distance from the stem position
+    def calculateHeightFromDistance(self, stem_height, crown_radius, distance):
+        bools = crown_radius[np.newaxis, np.newaxis, :] > distance
+        idx = np.where(bools)
+        height = np.full_like(distance, fill_value=-99999)
+        height[idx] = stem_height[idx[2]] + (4 * crown_radius[idx[2]]**2 -
+                                             distance[idx]**2)**0.5
+        return height, bools
+
+    ## This function reads x- and y-domain and mesh resolution
+    #  from project file and creates the mesh.\n
+    #  @param Tags to define plot size and spatial resolution: see tag documentation
+    def makeGrid(self, args):
+        missing_tags = [
+            "type", "domain", "x_1", "x_2", "y_1", "y_2", "x_resolution",
+            "y_resolution"
+        ]
+        for arg in args.iterdescendants():
+            tag = arg.tag
+            if tag == "x_resolution":
+                x_resolution = int(arg.text)
+            if tag == "y_resolution":
+                y_resolution = int(arg.text)
+            elif tag == "x_1":
+                x_1 = float(arg.text)
+            elif tag == "x_2":
+                x_2 = float(arg.text)
+            elif tag == "y_1":
+                y_1 = float(arg.text)
+            elif tag == "y_2":
+                y_2 = float(arg.text)
+            try:
+                missing_tags.remove(tag)
+            except ValueError:
+                raise ValueError(
+                    "Tag " + tag +
+                    " not specified for above-ground grid initialisation!")
+        if len(missing_tags) > 0:
+            string = ""
+            for tag in missing_tags:
+                string += tag + " "
+            raise KeyError(
+                "Tag(s) " + string +
+                "are not given for above-ground grid initialisation in project file."
+            )
+        l_x = x_2 - x_1
+        l_y = y_2 - y_1
+        x_step = l_x / x_resolution
+        y_step = l_y / y_resolution
+        self.min_r_crown = np.mean([x_step, y_step]) * 2**0.5
```

Comment:
Isn't half the size sufficient to at least get one node?
I would rather use np.ax, cause it's more robust

`
```suggestion
        self.min_r_crown = np.max([x_step, y_step]) * 1/2**0.5
```

`

Reply:
I thought we want at least 3?
and shouldn't it be np.min, then?

Reply:
Why is 3 better than one?
np.max in order to capture the largest spacing. Imagine x_step = 0.1 and y_step = 10. In that case we a min crown radius of 0.1/sqrt(2) does not imply a minimum of 1 (or 3) gridpoints.